xbmc/network/DNSNameCache.cpp

```diff
@@ -84,20 +109,42 @@ bool CDNSNameCache::Lookup(const std::string& strHostName, std::string& strIpAdd
 #endif
 
   // perform dns lookup
-  struct hostent *host = gethostbyname(strHostName.c_str());
-  if (host && host->h_addr_list[0])
+  struct addrinfo hints;
+  struct addrinfo *result(NULL);
+
+  memset(&hints, 0, sizeof(struct addrinfo));
+  int err;
+
+  // prefer DNS record type (A vs AAAA)  be this of active interface(address)
+  // otherwise (by default) system prefers A(IPv4) records and this will make
+  // troubles on dual configured hosts - but without wide area access over IPv4
+  do {
+    freeaddrinfo(result);
```

Comment:
I tested specifically both BSD(on osx) and Linux implementations (with same code) and via valgrind for mem pool development and it is fine. I was interested if it survives multiple freeing of the same pointer and valgrind set to detailed mem analysing found no single issue. 

The linux nor BSD documentation says anything on this topic, only a short desc - freeaddrinfo frees memory previously allocated on getaddrinfo() call.
So I wasn't still sure as it can work with current glibc but must not next year (when not specifically documented).

But then it somehow made sense after rereading the docs which defines a NULL (as the pointer returned from getaddrinfo()) being an OK result. example code is testing NULL on using the result, but not on freeing it. Basically the getaddrinfo()/freeaddrinfo() is very much following getifaddrs()/freeifaddrs() approach which is used for a long time here https://github.com/xbmc/xbmc/blob/master/xbmc/network/linux/NetworkLinux.cpp#L373 and https://github.com/xbmc/xbmc/blob/master/xbmc/network/linux/NetworkLinux.cpp#L396 .

Honestly, despite all that being said, if you ask for NULL test I won't protest.